### ISSUE TITLE
job().get() crashes on invalid characters in job scripts

### ISSUE THREAD

**Issue by mstud** [PySlurm/pyslurm#71]
Since PySlurm seems to include SHOW_DETAIL2 to get the batch script for each job, it throws an exception if it encounters certain characters in one of the users batch scripts, namely in function stringOrNone in slurm.pxd line 56 when it tries to decode the value using utf-8.
My workaround was to change the value.decode('utf-8') in there to a value.decode('utf-8', 'replace').

**Comment by giovtorres** [comment]
Yes,  I did change it to SHOW_DETAIL2 to get the batch script.  So, you prefer to replace the errors with question marks and not to ignore them?

**Comment by giovtorres** [comment]
What version of PySlurm and Slurm are you using?  Also, which special characters are you using?

I ran a job with the following sbatch directive on Slurm 15.08.11, Python 2.7, and PySlurm 15.08.8:
```
 #SBATCH --job-name=sleepíñg
```

PySlurm reports it as the following in the dictionary output:

```
 u'#SBATCH --job-name=sleep\xed\xf1g\n
```

I don't mind making this change, just want to replicate it first.  

Thanks,
Giovanni

**Comment by mstud** [comment]
I'm using PySlurm 15.08.8 with SLURM 15.08.10. Everything is fine as long as the batch script is saved as a UTF-8 file, but some of our users use Windows editors and save their scripts as iso-8859-1. The culprit characters were German umlauts, so an example could be:

#!/bin/bash
# für dich
sleep 1000

(Note the the "ü" in the comment)
Save it as latin-1, and you will get a UnicodeDecodeError exception if you are just using .decode('utf-8') without the _errors_ argument. Sure, you could also set errors to "ignore" so the umlauts will get filtered out, but I prefer "replace", because then you at least see that there is supposed to be character that just couldn't get decoded properly.

**Comment by giovtorres** [comment]
Fixed in 509cd68.  Thanks for pointing this out and suggesting the fix.
